pkg/grpc/client.go

```diff
@@ -0,0 +1,76 @@
+package grpc
+
+// gRPC client implementation to be able to connect to the gRPC server and perform the following operations:
+// - Start a remote JupyterLab server
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/cli/cli/v2/pkg/grpc/jupyter"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/metadata"
+)
+
+const (
+	requestTimeout = 30 * time.Second
+)
+
+type GrpcClient struct {
+	conn          *grpc.ClientConn
+	token         string
+	jupyterClient jupyter.JupyterServerHostClient
+}
+
+func New() *GrpcClient {
+	return &GrpcClient{}
+}
+
+// Connects to the gRPC server on the given port
+func (g *GrpcClient) Connect(ctx context.Context, port int, token string) error {
+	// Attempt to connect to the given port
+	conn, err := grpc.Dial(fmt.Sprintf("localhost:%d", port), grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithBlock())
```

Comment:
Should this be `127.0.0.1`?

Reply:
`localhost` seems to be working fine but it seems as though `127.0.0.1` is used everywhere else so I'll update this to do the same

Reply:
I forget when/where/why we decided to use 127.0.0.1 in the liveshare stuff but I remember there was a reason 😄 

I think there's another place where we use `localhost` in this PR, we should update that too.